corehq/apps/es/mappings/user_mapping.py

```diff
@@ -117,7 +117,7 @@
         },
         "domain_membership": {
             "dynamic": False,
-            "type": "object",
+            "type": "nested",
```

Comment:
Yes, that is the next thing I was going to do!

Reply:
[9e7b4095059703af19b35b1ae0711eb256d1dea3](https://github.com/dimagi/commcare-hq/pull/33726/commits/9e7b4095059703af19b35b1ae0711eb256d1dea3)

Reply:
I am wondering whether this change is actually worth it given the confusion it creates over how the couch and ES docs differ for mobile users doesn't seem justified for the little code cleanup it offers. What do you think?

Reply:
That's a fair consideration.  I'm hoping to bring things closer together to avoid accidentally introducing code that works for only one type of user.  For what it's worth, the `CommCareUser` class [already has `domains` and `domain_memberships` properties of this nature](https://github.com/dimagi/commcare-hq/blob/d9e67f9fde452ac5b3682af839f5b4583404436e/corehq/apps/users/models.py#L673-L679) for presumably the same reason.

The cleanup I'm referring to is stuff like this:
https://github.com/dimagi/commcare-hq/blob/d9e67f9fde452ac5b3682af839f5b4583404436e/corehq/apps/es/users.py#L199-L207

Sure it's not hard to work around in code, just seems annoying and kind of silly.  Though I do agree that it's nice to have the couch and ES docs similar in general.  Another possibility is to do both.  That is, we could send mobile workers with both `domain_membership` and `domain_memberships`, but only have the latter indexed in the mapping.  Then we can have data indexed in a way that's straightforward to query, but still have the expected fields populated in the return value.

I don't know, maybe we should invite comment from the broader team?